### PR TITLE
feat KPI-227 improve user page pagination

### PR DESCRIPTION
**Pull request by FreddyJR1995**
## Jira Link

[Go to Jira](https://kpinetwork.atlassian.net/browse/KPI-227)

## Types of changes

- [ ] Hotfix (Bug detected)
- [x] Feature (Issue from an Active Sprint)

## Checklist:

- [x] Unit Test.
- [x] Coverage.

### CODE REVIEW

File: infrastructure/policy/iam.tf [hunk 1]
```diff
@@ -1159,7 +1159,8 @@ resource "aws_iam_role_policy" "authorize_lambda_get_users_and_groups" {
       "Effect": "Allow",
       "Action": [
         "cognito-idp:AdminListGroupsForUser",
-        "cognito-idp:ListUsers"
+        "cognito-idp:ListUsers",
+        "cognito-id:ListUsersInGroup"
```
Comment (kellycastrof) [inline infrastructure/policy/iam.tf:1163]: I think this should be codnito-idp instead of cognito-id

File: src/service/users/users_service.py [hunk 2]
```diff
@@ -28,14 +30,13 @@ def set_users_groups(self, users, user_pool_id) -> list:
...
-    def get_users(self, user_pool_id: str, limit: int = 10, token: str = None) -> dict:
-        params = self.get_users_params(user_pool_id, limit, token)
-        response = self.client.list_users(**params)
+    def get_users(self, user_pool_id: str, limit: int, token: str, group: str) -> dict:
+        params = self.get_users_params(user_pool_id, limit, token, group)
+        response = self.client.list_users_in_group(**params)
         users = self.response_user.proccess_users(response.get("Users", []))
-
-        self.set_users_groups(users, user_pool_id)
+        print("token", response)
```
Comment (kellycastrof) [inline src/service/users/users_service.py:37]: Remove this

File: infrastructure/network/gateways.tf [hunk 3]
```diff
@@ -433,6 +433,7 @@ resource "aws_api_gateway_method" "get_users_method" {
   request_parameters = {
     "method.request.querystring.token" = false
     "method.request.querystring.limit" = false
+    "method.request.querystring.group" = false
```
Comment (kellycastrof) [inline infrastructure/network/gateways.tf:436]: This should be mandatory

**Review by kellycastrof — CHANGES_REQUESTED** [review]

**Review by dsgarcia8 — CHANGES_REQUESTED** [review]

File: src/handlers/users/get_users_handler.py [hunk 4]
```diff
@@ -12,12 +12,16 @@ def handler(event, _):
         users_service = get_users_service_instance(event, logger)
         limit = 10
         token = ""
+        group = "demo_customer_group"
```
[the remark this answers is not in the document]
Comment (FreddyJR1995) [reply src/handlers/users/get_users_handler.py]: variable added and value changed

**Review by dsgarcia8 — APPROVED** [review]

**Review by kellycastrof — APPROVED** [review]

**Pull request merged**